src/broker/module.c

```diff
@@ -91,6 +91,8 @@ static int setup_module_profiling (module_t *p)
     cali_begin_int_byname ("flux.rank", p->rank);
     cali_begin_string_byname ("flux.name", p->name);
 #endif
+    // Set the name of each thread to its module name
+    pthread_setname_np(pthread_self(), p->name);
```

Comment:
Wow, great idea! I've definitely hit the "what module am I looking at here?" issue in gdb.

Very minor comments:

 - space between function name and opening parens for consistency (sorry about that)
 - check return code or cast to  `(void)` to establish obvious intent to ignore return code

Reply:
Yeah, I didn't realize this was possible - excellent change.

Reply:
Ok, made those adjustments and in double-checking the return codes also found that there's a length limit to worry about, so we now check the length and if it's too long take the prefix.